crates/common/tedge_config/src/lib.rs

```diff
@@ -7,21 +7,34 @@ pub use self::tedge_config_cli::error::*;
 pub use self::tedge_config_cli::models::*;
 pub use self::tedge_config_cli::tedge_config::*;
 pub use self::tedge_config_cli::tedge_config_location::*;
-pub use self::tedge_config_cli::tedge_config_repository::*;
 pub use camino::Utf8Path as Path;
 pub use camino::Utf8PathBuf as PathBuf;
 pub use certificate::CertificateError;
 pub use tedge_config_macros::all_or_nothing;
 pub use tedge_config_macros::OptionalConfig;
 
-/// loads the new tedge config from system default
-pub fn get_new_tedge_config() -> Result<TEdgeConfig, TEdgeConfigError> {
-    let tedge_config_location = TEdgeConfigLocation::default();
-    TEdgeConfigRepository::new(tedge_config_location).load()
-}
+impl TEdgeConfig {
+    pub fn new(config_location: TEdgeConfigLocation) -> Result<Self, TEdgeConfigError> {
+        config_location.load()
+    }
+
+    pub fn load(config_dir: &Path) -> Result<TEdgeConfig, TEdgeConfigError> {
```

Comment:
Wondering if we could just merge the `new` and this `load` functions into a single one that takes an `AsRef<Path>` as the argument, with an `impl AsRef<Path> for TEdgeConfigLocation`.

Reply:
This was my initial plan. Unfortunately, this is more involved as in many places both the config and its location are used. We have first to simplify the configuration of misc actors and plugins (notably in the c8y mapper).